### ISSUE TITLE
Is this abandoned? Are there lessons learned?

### ISSUE THREAD

**Issue by robmarkcole** [Clay-foundation/stacchip#11]
As title, thanks @brunosan

**Comment by brunosan** [comment]
@yellowcap used it for another project and we (Clay or Lgnd) ended up not needing it anymore, so we are not using it or actively maintaining open issues.

**Comment by yellowcap** [comment]
Similar, not using it at the moment, so yes its not actively maintained anymore

In terms of lessons learned, for me the approach worked really well overall when we used it for Clay. 

## For training data

We used this technique to do prepare tiles on AWS batch at very large scale and then bundled up the tiles into numpy zip files that we then fed into the training data loader. We optimized how many chips we bundled into one zip package so we could feed the data as fast as possible to the model. We randomized the tiles that we put into each bundle, so we could just run through them without needing to shuffle things at training time. 

Overall the main idea of creating and index first and then doing the actual chipping based on sampling from the indexes worked nicely. I would do something similar again.

## For inference

For inference at scale we also used AWS batch, but actually used the streaming approach. So we used stacchip to dynamically create chips for each scene we were processing, and streamed this to the model for inference. We did not persist those chips.

As an example, this approach allowed us to process 1 year of 1m NAIP imagery for all of the US relatively cheap and fast.

**Comment by robmarkcole** [comment]
Thanks @yellowcap RE streaming did you ever look into https://github.com/Lightning-AI/litData ?

**Comment by yellowcap** [comment]
At the time we looked at a bunch of streaming options but did not find any that were really fast enough. If I remember the pytorch loader was doing parallel batch loading but only at the end/beginning of each batch. So in our case that would be on the order of 1-2 min of GPU downtime. It did not pre-load the data while training on a current batch. That is why we went for a custom bundling where 1 batch was roughly one pre-randomized file. Cc @srmsoumya
